spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/adapter/MessagingMessageListenerAdapter.java

```diff
@@ -149,7 +149,9 @@ public void onMessage(org.springframework.amqp.core.Message amqpMessage, Channel
 							.build();
 					Object errorResult = this.errorHandler.handleError(amqpMessage, message, e);
 					if (errorResult != null) {
-						handleResult(new InvocationResult(errorResult, null, null), amqpMessage, channel, message);
+						handleResult(new InvocationResult(errorResult,
+									this.handlerAdapter.getSendToForPayload(message.getPayload()), null),
```

Comment:
The Type send here is still null. 
Do you think it would work better with POJOs, If type is send here ?

Reply:
Good point, although it's only important for complex return types with lots of generics; for simple POJOs the actual return value is used.

But I will change it.

Reply:
Wonderful but I have a question here:
For POJOs where jackson mapper is being used, a Type_Id is attached to header which is usually the POJO name with its namespace, Does that relate with type here ?

Reply:
For simple classes the `result.getClass()` is enough for the type ID.

For more complex types, eg. `Map<Foo, Bar>` we need the generic return type so that information is conveyed in other headers.

Of course, the error handler must return the same type as the listener, otherwise the headers won't match (the generic return type of the error handler is `Object`, so we can't use that.